components/textfield/index.css

```diff
@@ -131,207 +460,233 @@ governing permissions and limitations under the License.
+  }
+
+  /****** Input Disabled 🚫 ******/
+  .spectrum-Textfield.is-disabled &,
+  .spectrum-Textfield.is-disabled:hover &,
+  &:disabled {
+    background-color: var(--mod-textfield-background-color-disabled, var(--spectrum-textfield-background-color-disabled));
+    border-color: transparent;
+    color: var(--highcontrast-textfield-text-color-disabled, var(--mod-textfield-text-color-disabled, var(--spectrum-textfield-text-color-disabled)));
 
-    /* removes the side padding to align the text properly */
-    padding-inline-start: var(--spectrum-textfield-quiet-texticon-padding-left);
-    padding-inline-end: var(--spectrum-textfield-quiet-texticon-padding-right);
+    /* For safari mobile browser */
+    -webkit-text-fill-color: var(--highcontrast-textfield-text-color-disabled, var(--mod-textfield-text-color-disabled, var(--spectrum-textfield-text-color-disabled)));
 
-    /* Treat all quiet inputs and textareas the same */
+    /* Disable the resize functionality when disabled */
     resize: none;
-    overflow-y: hidden;
 
-    .is-invalid& {
-      padding-inline-end: calc(
-        var(--spectrum-textfield-texticon-invalid-icon-width) +
-          var(--spectrum-textfield-quiet-texticon-invalid-icon-margin-left)
-      );
-    }
-    .is-valid& {
-      padding-inline-end: calc(
-        var(--spectrum-textfield-texticon-success-icon-width) +
-          var(--spectrum-textfield-quiet-texticon-success-icon-margin-left)
-      );
+    /* The opacity must be set to 1 */
+    opacity: 1;
+
+    &::placeholder {
+      color: var(--highcontrast-textfield-text-color-disabled, var(--mod-textfield-text-color-disabled, var(--spectrum-textfield-text-color-disabled)));
     }
   }
-}
-
-.spectrum-Textfield-validationIcon {
-  position: absolute;
-  pointer-events: all;
 
+  /****** Input - Quiet 🤫 ******/
   .spectrum-Textfield--quiet & {
-    padding-inline-end: 0;
-  }
+    border-block-start-width: 0;
+    border-inline-width: 0;
+    margin-block-end: var(--mod-textfield-spacing-block-quiet, var(--spectrum-textfield-spacing-block-quiet));
+    padding-inline: var(--mod-textfield-spacing-inline-quiet, var(--spectrum-textfield-spacing-inline-quiet));
+    background-color: transparent;
+    border-radius: 0;
 
-  .spectrum-Textfield.is-invalid & {
-    inline-size: var(--spectrum-textfield-texticon-invalid-icon-width);
-    block-size: var(--spectrum-textfield-texticon-invalid-icon-height);
-    inset-block-end: calc(
-      calc(var(--spectrum-textfield-texticon-height) / 2) -
-        calc(var(--spectrum-textfield-texticon-invalid-icon-height) / 2)
-    );
+    /* Treat all quiet inputs and text areas the same */
+    resize: none;
+    overflow-y: hidden;
+   }
 
-    inset-inline-end: var(--spectrum-textfield-icon-inline-end-override, var(--spectrum-textfield-texticon-invalid-icon-margin-left));
+  /****** Input - Quiet 🤫 + Disabled 🚫 ******/
+  .spectrum-Textfield--quiet.is-disabled &,
```

Comment:
Just curious, why use both a class and the :disabled selector?

Reply:
I'm assuming she did this just to cover any outlier use cases of just `:disabled`. Would you all prefer us to use classes over the attributes?

Reply:
I think we've been inconsistent with what we've used and when. I think we can safely use just the `:disabled` pseudo-class on this component. We should note that we're removing the `.is-disabled` class in the Migration Notes section of the docs, though.

Reply:
@pfulton I think we still need to have the class applied on the wrapper div so that the other elements (character count, help text, label) get the proper styling when the field is disabled, since we can't add a `disabled` attribute to a div. So I don't know if the class should be removed. I guess the issue is just do we need to add catches like this for the native pseudo-classes rather than just our classes?